packages/template-retail-react-app/app/static/dwac-21.7.js

```diff
@@ -0,0 +1,470 @@
+            }
+            else if (c == 32) {
+                encoded += '+';					// (space)
+            }
+            else if (c < 128) {
+                encoded += convertByte(c);
+            }
+            else if ((c >= 128) && (c < 2048)) {
+                encoded += convertByte((c >> 6) | 0xC0);
+                encoded += convertByte((c & 0x3F) | 0x80);
+            }
+            else if ((c >= 2048) && (c < 65536)) {
+                encoded += convertByte((c >> 12) | 0xE0);
+                encoded += convertByte(((c >> 6) & 0x3F) | 0x80);
+                encoded += convertByte((c & 0x3F) | 0x80);
+            }
+            else if (c >= 65536) {
+                encoded += convertByte((c >> 18) | 0xF0);
+                encoded += convertByte(((c >> 12) & 0x3F) | 0x80);
+                encoded += convertByte(((c >> 6) & 0x3F) | 0x80);
+                encoded += convertByte((c & 0x3F) | 0x80);
+            }
+        }
+
+        return encoded;
+    }
+
+    // Returns the value of the analytics cookie set on the server
+    function getAnalyticsCookieValue() {
+        var acceptFunction = function(name) {
+            return (name.length > 5) && (name.substring(0, 5) === 'dwac_');
+        };
+        return getCookieValue(acceptFunction);
+    };
+
+    // Contextual information retrieved from the server
+    var analyticsContext = (function() {
+        if (dw.ac._analytics_enabled === "false") {
+            return {
+                enabled: false,
+                dwEnabled: false
+            };
+        }
+        var cookieValue = getAnalyticsCookieValue();
+        if (cookieValue == null) {
+            return {
+                visitorID: '__ANNONYMOUS__',
+                customer: '__ANNONYMOUS__',
+                siteCurrency: '',
+                sourceCode: '',
+                enabled: "true",
+                timeZone: dw.ac._timeZone,
+                dwEnabled: "true",
+                encoding: 'ISO-8859-1'
+            };
+        }
+
+        var tokens = cookieValue.split('|');
+
+        return {
+            visitorID: tokens[0],
+            repository: tokens[1],
+            customer: tokens[2],
+            sourceCode: tokens[3],
+            siteCurrency: tokens[4],
+            enabled: tokens[5] == "true",
+            timeZone: tokens[6],
+            dwEnabled: tokens[7] == "true",
+            encoding: 'ISO-8859-1'
+        };
+    }());
+
+    // Turn debugging on or off
+    var setDebugEnabled = function(enabled) {
+        if (typeof enabled != 'boolean') {
+            return;
+        }
+
+        setCookieValue('dwacdebug', '' + enabled, -1);
+    };
+
+    // Returns true if debug is enabled, false otherwise
+    function isDebugEnabled() {
+        var acceptFunction = function(name) {
+            return name === 'dwacdebug';
+        };
+        return getCookieValue(acceptFunction) === 'true';
+    };
+
+    // Map data structure
+    function Map() {
```

Comment:
`Map` is a reserved function name (e.g. keyword) in ECMAScript... are you intentionally overloading it here? Why not use a new function and avoid the potential risk of unexpected side-effects that might come with overloading base web API functionality in a way that is potentially not obvious to the user who is building a PWA Kit project?